tests/testthat/test-check.R

```diff
@@ -0,0 +1,109 @@
+test_that("check_string_param", {
+  expect_no_error(check_string_param("name", "value"))
+  expect_no_error(check_string_param("name", "value", required = FALSE))
```

Comment:
praise: I like the way this reads with the `required = FALSE` spelled out.  (And also like the `expect_no_error` over the old arcane way of `expect_error({expr}, regexp = NA)` or whatever it was.)